### PR TITLE
TG-964 Upgrade docker-compose with healthchecks handler

### PR DESCRIPTION
**Pull request by daniele20tab**

### CODE REVIEW

File: {{cookiecutter.project_dirname}}/.env_template [hunk 1]
```diff
@@ -14,13 +14,18 @@ DJANGO_SUPERUSER_PASSWORD={{ cookiecutter.project_slug }}
 DJANGO_SUPERUSER_USERNAME=${USER}
 EMAIL_URL=console:///
 PYTHONBREAKPOINT=IPython.core.debugger.set_trace
+# PYTHONDEVMODE=1
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/.env_template:17]:
```suggestion
PYTHONDEVMODE=""
```

Comment (niccolomineo) [reply {{cookiecutter.project_dirname}}/.env_template:17]: Just curious: how come setting `PYTHONDEVMODE` to `0` wouldn't work?

Comment (trottomv) [reply {{cookiecutter.project_dirname}}/.env_template:17]: setting `PYTHONDEVMODE` to `0` would be preferable, but I checked locally on some ongoing project and I realized that unfortunately it doesn't work

Comment (niccolomineo) [reply {{cookiecutter.project_dirname}}/.env_template:17]: > @niccolomineo we can add also `PYTHONTRACEMALLOC=0` in this PR

Just go ahead and add it if you wish 👍

Comment (niccolomineo) [reply {{cookiecutter.project_dirname}}/.env_template:17]:
```suggestion
PYTHONDEVMODE=""
PYTHONTRACEMALLOC=0
```

File: {{cookiecutter.project_dirname}}/docker-compose.yaml [hunk 2]
```diff
@@ -29,29 +29,38 @@ services:
...
+        USER_ID: ${USER_ID:-1000}
+        USER: ${USER:-appuser}
       context: ./{{ cookiecutter.frontend_service_slug }}
-      dockerfile: docker/local.Dockerfile{% if cookiecutter.backend_type != "none" %}
-    depends_on:
-      {{ cookiecutter.backend_service_slug }}:
-        condition: service_started{% endif %}
+      dockerfile: ${FRONTEND_DOCKER_FILE:-docker/local.Dockerfile}{% if cookiecutter.backend_type != "none" %}
+    healthcheck:
+      test: wget -O- -q http://localhost:3000/api/health/ || exit 1
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/docker-compose.yaml:47]:
```suggestion
      test: ["CMD", "wget", "--no-verbose", "--tries=1", "--spider", "http://localhost:3000/api/health/", "||", "exit", "1"]
      interval: 5s
      timeout: 3s
      start_period: 30s
      retries: 5
```

File: {{cookiecutter.project_dirname}}/docker-compose.yaml [hunk 3]
```diff
@@ -29,36 +29,55 @@ services:
...
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/docker-compose.yaml:52]: Would be useful to expose the health check endpoint in order to have a valid enpoint available for active monitoring of the service. The path of the frontend could overlap with that of the backend, I would find it useful to find a different but fairly symmetrical path between the frontend and backend services.

For example an idea could be to obtain this result:
https://projectdomain.com/frontend/health/
https://projectdomain.com/backend/health/

**Comment by pauloxnet** [comment]
Please add a code to the related US/issue

File: {{cookiecutter.project_dirname}}/docker-compose.yaml [hunk 4]
```diff
@@ -29,36 +29,61 @@ services:
       - PACT_BROKER_URL
       - PACT_PROVIDER_NAME{% endif %}
       - PYTHONBREAKPOINT
+      - PYTHONDEVMODE
+      - PYTHONTRACEMALLOC
+    healthcheck:
+      test: curl --fail --head http://backend:8000/{{ cookiecutter.backend_service_slug }}/health/ || exit 1
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/docker-compose.yaml:35]: modificare l'url con il valore cookiecutterato

Comment (trottomv) [inline {{cookiecutter.project_dirname}}/docker-compose.yaml:35]: aggiungere curl sul Dockerfile del backend

**Review by pauloxnet — APPROVED** [review]

**Pull request merged**